### DISCUSSION TITLE
sort class by names variables, functions

### DISCUSSION THREAD

**Discussion by YesidMelo** [detekt/detekt#7527] [Q&A] [answered]
Hi, guys I have been applying the coding standard for my projects using your tool, which I thank you for creating.

However I have tried to apply a bit of order in the structure of the classes and files to facilitate the search inside the android studio ide in the following way

class MyClass {
  //region variables
  lateinit var var1 
  private val var2 = 0
  private var var3 = 0
  //endregion

  //region override methods
  override fun b1() {}
  override fun b2() {}
  override fun b3() {}
  //endregion

  //region public methods
  private fun a1() {}

  private fun a2(
      param1: String,
      param2: String, 
      param3: String
  ) {}
  //endregion

 //region private methods

  //endregion
}

is there a way to check for compliance with the standard alphabetical order of variables, public functions, private functions and inherited functions currently in the library. could this be developed somehow?

Thanks guys for such a good tool.

PDA
can you guys update the documentation to implement detetk in projects that use .toml files i have been having trouble adapting the library to recent projects thanks in advance.

**Comment by BraisGabin** [comment, marked as answer]
That rule could be implemented. But that should be developed as a third party rule. I'm general is advices to sort the functions in a logical way. Creating clusters of related functions instead of alphabetical. But each team have different needs so, yes, what you say can be implemented.

About the documentation. Could you be more precise about what's your problem? And, did you solved it already? If that's the case we welcome PRs to improve our documentation.

**Reply by YesidMelo** [reply]
Of course, check out this android documentation, https://developer.android.com/build/migrate-to-catalogs,  if I'm not wrong it is centralising the use of dependencies, but I can't make use of the library's own elements like detektGenerateConfig.
